Podcast/RecommendedEpisodesOuterTableViewCell.swift

```diff
@@ -50,19 +52,38 @@ class RecommendedEpisodesOuterTableViewCell: UITableViewCell, UITableViewDelegat
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return dataSource?.numberOfRecommendedEpisodes(forRecommendedEpisodesOuterTableViewCell: self) ?? 0
+        if let numEpisodes = dataSource?.numberOfRecommendedEpisodes() {
+            return max (1, numEpisodes)
+        }
+        else { return 0 }
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as? EpisodeTableViewCell else { return UITableViewCell() }
-        let episode = dataSource?.recommendedEpisodesTableViewCell(cell: self, dataForItemAt: indexPath) ?? Episode()
-        cell.setupWithEpisode(episode: episode)
-        cell.delegate = self
-        return cell
+        guard let numEpisodes = dataSource?.numberOfRecommendedEpisodes() else { return UITableViewCell() }
+        if numEpisodes > 0 {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as? EpisodeTableViewCell else { return UITableViewCell() }
+            let episode = dataSource?.recommendedEpisodesTableViewCell(dataForItemAt: indexPath) ?? Episode()
+            cell.setupWithEpisode(episode: episode)
+            cell.delegate = self
+            return cell
+        }
+        else {
+            guard let user = dataSource?.getUser() else { return UITableViewCell() }
+            
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "NullCell", for: indexPath) as? NullProfileTableViewCell else { return NullProfileTableViewCell() }
+            cell.setUp(user: user)
+            return cell
+        }
     }
     
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return UITableViewAutomaticDimension
+        guard let numEpisodes = dataSource?.numberOfRecommendedEpisodes() else { return UITableViewAutomaticDimension }
```

Comment:
isn't numEpisodes an optional? so we can just do `if let _ =  dataSource?.numberOfRecommendedEpisodes() {return UITableViewAutomaticDimension } else { return 108}`

Reply:
but put the constant at the top